packages/types/src/documents/app/row.ts

```diff
@@ -1,22 +1,76 @@
 import { Document } from "../document"
 
 export enum FieldType {
+  // a primitive type, stores a string, called Text within Budibase. This is one of the default
+  // types of Budibase, if an external type is not fully understood, we will treat it as text.
   STRING = "string",
+  // similar to string type, called Long Form Text within Budibase. This is mainly a frontend
+  // orientated type which disables a larger text input area. This can also be used
+  // in conjunction with the 'useRichText' option to support a markdown editor/viewer.
   LONGFORM = "longform",
+  // similar to string type, called Options within Budibase. This works very similarly to
+  // the string type within the backend, but is validated to a list of options. This will
+  // be displayed a select input within the builder/client.
   OPTIONS = "options",
+  // a primitive type, stores a number, as a floating point, called Number within Budibase.
+  // this type will always represent numbers as reals/floating point - there is no integer only
+  // type within Budibase.
   NUMBER = "number",
+  // a primitive type, stores a boolean, called Boolean within Budibase. This is often represented
+  // as a toggle or checkbox within forms/grids.
   BOOLEAN = "boolean",
+  // a JSON type, this type is always an array of strings, called Multi-select within Budibase.
+  // This type can be compared to the options type, as it functions similarly, but allows picking
+  // multiple options rather than a single option.
   ARRAY = "array",
```

Comment:
If this is a JSON type does that mean it is represented as a string column in some data sources?

Reply:
This will always be represented as a JSON type if it is available however if there is no JSON type it will default to a string. There are two cases where this happens:
1. Couch - we supply stringified JSON to Couch, but it is represented as JSON for Lucene to search.
2. SQS - SQLite doesn't have a JSON type, so we will represent it as a string or blob here.